src/BlockTypePalette.cpp

```diff
@@ -0,0 +1,80 @@
+#include "Globals.h"
+#include "BlockTypePalette.h"
+
+
+
+
+BlockTypePalette::BlockTypePalette()
+{
+	// Nothing needed yet
+}
+
+
+
+
+
+UInt32 BlockTypePalette::index(const AString & aBlockTypeName, const BlockState & aBlockState)
```

Comment:
I'm not really sure if `index` is a right name if it can also append the blockstate when it's not there yet. I'm not really fond of the alternatives I came up with though:

- indexOrAdd
- findIndexOrAdd
- indexOfOrAdd
- safeIndexOf

Reply:
I think this is the most common operation, you want the index of a blockspec and you don't care if it's there or not. Basically it's the same as `std::map::operator[]`.

I think it's more understandable to have the pair `index` and `maybeIndex`, than `indexOrAdd` and `indexOrFail`. Still, I'm open to suggestions, too.

Reply:
Lets keep it like this for now.